### PR TITLE
Add docker build

### PR DESCRIPTION
**Pull request by galal-hussein**

### CODE REVIEW

**Review by iabdalkader — CHANGES_REQUESTED** [review]
Please move all new files to openmv/docker/ so it's all contained in there. You'll need to update the build script, gitignore and README, the layout/filenames should look like this:
```
openmv/docker/Makefile
openmv/docker/dockerfile
openmv/docker/build.sh
```
Note: You should probably set a `TOP_DIR` variable to `../` (in Makefile or the build script) and you can use `-C` with git to run the command in a different dir, for example:

```
TOP_DIR = ../
git -C $(TOP_DIR)/src/micropython/ submodule update --init --depth=1
```

Do we really need the build script ? If not you can just put the commands in the Makefile.

File: scripts/build [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -e -x
+
+cd $(dirname $0)/..
+
+
+export PATH=/source/gcc/gcc/bin:/source/cmake/cmake/bin:$PATH
+git submodule update --init --depth=1
+git -C src/micropython/ submodule update --init --depth=1
```
[the remark this answers is not in the document]
Comment (galal-hussein) [reply scripts/build]: fixed

File: src/README.md [hunk 2]
```diff
@@ -112,6 +112,23 @@ After doing this OpenMV IDE will then be able to read the newly compiled binarie
 
 To update your OpenMV Cam's firmware you want to load the `firmware.bin` file onto your OpenMV Cam using the `Tools->Run Bootloader` command in OpenMV IDE. Please make sure to only load firmware meant for your model of the OpenMV Cam onto it. The IDE does not check this. If you accidentally load the wrong firmware your OpenMV Cam will just crash without being damaged when trying to execute the main firmware image and you can then just use the bootloader onboard again to load the correct `firmware.bin`.
 
+### Docker Build
+
+To build the firmware using docker, follow the following steps:
+
+```
+git clone https://github.com/openmv/openmv.git
```
[the remark this answers is not in the document]
Comment (galal-hussein) [reply src/README.md]: fixed

File: src/README.md [hunk 3]
```diff
@@ -112,6 +112,23 @@ After doing this OpenMV IDE will then be able to read the newly compiled binarie
...
+make TARGET=<TARGET NAME>
+```
+
+After building you should see the target build output under `build/<TARGET_NAME>`, for example
+
+```
+ls build/OPENMV4 
+bootloader.bin  bootloader.dfu  bootloader.elf  firmware.bin  firmware.dfu  firmware.elf  openmv.bin  uvc.bin  uvc.dfu  uvc.elf
+```
+
```
[the remark this answers is not in the document]
Comment (galal-hussein) [reply src/README.md]: fixed

**Review by iabdalkader — APPROVED** [review]

**Pull request merged**